src/udev/dmi_memory_id/dmi_memory_id.c

```diff
@@ -145,123 +130,136 @@ static void dmi_print_memory_size(
 
-static const char *dmi_memory_array_location(uint8_t code) {
+static void dmi_memory_array_location(uint8_t code) {
         /* 7.17.1 */
         static const char *location[] = {
                 [0x01] = "Other",
-                "Unknown",
-                "System Board Or Motherboard",
-                "ISA Add-on Card",
-                "EISA Add-on Card",
-                "PCI Add-on Card",
-                "MCA Add-on Card",
-                "PCMCIA Add-on Card",
-                "Proprietary Add-on Card",
-                "NuBus" /* 0x0A */
+                [0x02] = "Unknown",
+                [0x03] = "System Board Or Motherboard",
+                [0x04] = "ISA Add-on Card",
+                [0x05] = "EISA Add-on Card",
+                [0x06] = "PCI Add-on Card",
+                [0x07] = "MCA Add-on Card",
+                [0x08] = "PCMCIA Add-on Card",
+                [0x09] = "Proprietary Add-on Card",
+                [0x0A] = "NuBus",
         };
         static const char *location_0xA0[] = {
-                "PC-98/C20 Add-on Card",
-                "PC-98/C24 Add-on Card",
-                "PC-98/E Add-on Card",
-                "PC-98/Local Bus Add-on Card",
-                "CXL Flexbus 1.0" /* 0xA4 */
+                [0x00] = "PC-98/C20 Add-on Card",       /* 0xA0 */
+                [0x01] = "PC-98/C24 Add-on Card",       /* 0xA1 */
+                [0x02] = "PC-98/E Add-on Card",         /* 0xA2 */
+                [0x03] = "PC-98/Local Bus Add-on Card", /* 0xA3 */
+                [0x04] = "CXL Flexbus 1.0",             /* 0xA4 */
         };
+        const char *str = OUT_OF_SPEC_STR;
+
+        if (code < ELEMENTSOF(location) && location[code])
+                str = location[code];
+        else if (code >= 0xA0 && code < (ELEMENTSOF(location_0xA0) + 0xA0))
+                str = location_0xA0[code - 0xA0];
 
-        if (code < ELEMENTSOF(location) && code[location])
-                return location[code];
-        if (code >= 0xA0 && code < (ELEMENTSOF(location_0xA0) + 0xA0))
-                return location_0xA0[code - 0xA0];
-        return OUT_OF_SPEC_STR;
+        printf("MEMORY_ARRAY_LOCATION=%s\n", str);
 }
 
-static const char *dmi_memory_array_ec_type(uint8_t code) {
+static void dmi_memory_array_ec_type(uint8_t code) {
         /* 7.17.3 */
         static const char *type[] = {
                 [0x01] = "Other",
-                "Unknown",
-                "None",
-                "Parity",
-                "Single-bit ECC",
-                "Multi-bit ECC",
-                "CRC" /* 0x07 */
+                [0x02] = "Unknown",
+                [0x03] = "None",
+                [0x04] = "Parity",
+                [0x05] = "Single-bit ECC",
+                [0x06] = "Multi-bit ECC",
+                [0x07] = "CRC",
         };
 
-        if (code < ELEMENTSOF(type) && code[type])
-                return type[code];
-        return OUT_OF_SPEC_STR;
+        if (code != 0x03) /* Do not print "None". */
```

Comment:
Why? Wouldn't it be better to know that the property existed and was set, rather than not print it?

Reply:
You dropped that value previously. Now, `None` is also shown.

Reply:
Oh, I got this wrong then, sorry.